packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -502,6 +511,18 @@ export class Function extends FunctionBase {
     const code = props.code.bind(this);
     verifyCodeConfig(code, props.runtime);
 
+    let profilingGroupEnvironmentVariables = {};
+    if (props.profiling) {
+      this.profilingGroup = new ProfilingGroup(this, 'ProfilingGroup');
```

Comment:
I'm not familiar with CodeGuru profiling, but it feels we should also allow customers to specify their own profilingGroup as an option.

An additional optional property called `profilingGroup`. We can then create a new one if it's not specified.

Reply:
i'm divided about this, i'll leave it to you to decided. 

i wonder why a user would create a profiling group for their lambda outside the lambda setup, if at all. 
if users do need it in the future we can add then, i feel we should be conservative about the number of options we provide since CDK is okay with being opinionated.

Reply:
From [this page](https://docs.aws.amazon.com/codeguru/latest/profiler-ug/working-with-profiling-groups.html) -

> A profiling group is a set of applications that are profiled together as a unit. Application data is sent by the Amazon CodeGuru Profiler profiling agent to a single profile group. Data from all applications in a profiling group are aggregated and analyzed together.

And from [this page](https://docs.aws.amazon.com/codeguru/latest/profiler-ug/setting-up.html#setting-up-step-2) - 

> For example, if you have a collection of microservices that handle restaurant recommendations, you can collect profile data and identify performance issues across all these microservices in a single profiling group named "Restaurant-Recommendations".

Based on these suggestions, I'm now more inclined to provide this option.

Reply:
fair enough